### DISCUSSION TITLE
Can't interact with UI with network cable plugged in

### DISCUSSION THREAD

**Discussion by patmaddox** [helloSystem/hello#297] [General]
I'm trying to get started with 0.7 on a 2013 iMac. Something seems to not like my NIC (which uses `bge` driver according to `ifconfig`).

## Can't interact with UI when booting with network cable plugged in (doesn't finish booting?)

If I boot from USB with the network cable plugged in, it seems to not finish booting. I see the desktop, and an icon called `Starvolume`, but I don't see `Live` or `EFI` (which is what I see when I boot without a cable. The mouse and keyboard are non-responsive.

## Attaching network cable after booting makes UI non-interactive

If I boot from USB without the cable plugged in, I get a responsive UI and can open applications. I've opened the Terminal and looked at `ifconfig`. After I plug the network cable in, I can still use the Terminal and see that the NIC has been configured via DHCP.

Next I try to launch an application - `Create Live Media` so I can install helloSystem onto the internal hard drive. At this point I'm unable to interact with the UI at all. I can move the mouse cursor around, but not click anything, and the keyboard is unresponsive. I can no longer interact with Terminal either.

---

I don't have a clue how to debug this since I can't interact with anything after plugging in the network cable. Any ideas?

**Comment by patmaddox** [comment]
I've made some progress... I tried GhostBSD, and experienced the same behavior: running `dhclient` on the interface after plugging the cable in would make the UI unresponsive (other than the mouse, which I can still move).

I did some research and found that by adding `echo dev.bge.0.msi=0 >> /boot/loader.conf` I could run `dhclient` without freezing the UI.

This is where GhostBSD and helloSystem diverge. With GhostBSD, I can run `dhclient` and get on the network. With helloSystem, I can plug in the network cable without it freezing immediately - but then running either `ifconfig` or `dhclient` causes the interface to freeze (other than moving the mouse, still).

---

Edit: I don't have any such problems with using vanilla FreeBSD with XFCE.

**Comment by probonopd** [comment]
Thank you very much @patmaddox. 

It appears that without the workaround, the `dhclient` process does not complete, preventing from the boot process from completing.

Possibly this is related to
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=229727.

**Comment by probonopd** [comment]
I can reproduce the issue, have opened 
https://github.com/helloSystem/ISO/issues/355

**Reply by probonopd** [reply]
Unfortunately, I could not find a solution that works on the Macmini4,1 yet. Any help appreciated.
